### PR TITLE
Add latest wolftpm

### PR DESCRIPTION
**Pull request by tomoveu**
Updating the wolfTPM submodule to latest master and enabling WOLFTPM_USER_SETTINGS.

**this should be tested well before merging** due to the major version change v1.9->v2.0

### CODE REVIEW

**Review by danielinux — CHANGES_REQUESTED** [review]
Test `test-61-forward-update-no-downgrade-TPM` fails (link error)

```
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: lib/wolfTPM/src/tpm2.o: in function `TPM2_ResponseProcess.isra.3':
tpm2.c:(.text.TPM2_ResponseProcess.isra.3+0xdc): undefined reference to `TPM2_CalcRpHash'
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: tpm2.c:(.text.TPM2_ResponseProcess.isra.3+0xfe): undefined reference to `TPM2_CalcHmac'
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: tpm2.c:(.text.TPM2_ResponseProcess.isra.3+0x13e): undefined reference to `TPM2_ParamDec_CmdResponse'
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: lib/wolfTPM/src/tpm2.o: in function `TPM2_CommandProcess':
tpm2.c:(.text.TPM2_CommandProcess+0x140): undefined reference to `TPM2_ParamEnc_CmdRequest'
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: tpm2.c:(.text.TPM2_CommandProcess+0x166): undefined reference to `TPM2_CalcCpHash'
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: tpm2.c:(.text.TPM2_CommandProcess+0x18a): undefined reference to `TPM2_CalcHmac'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:107: wolfboot.elf] Error 1
make[2]: Leaving directory '/root/wolfboot'
make[1]: *** [tools/test.mk:178: test-01-forward-update-no-downgrade] Error 2
make[1]: Leaving directory '/root/wolfboot'
make: *** [tools/test.mk:312: test-61-forward-update-no-downgrade-TPM] Error 2
```

**Comment by tomoveu** [comment]
@danielinux updated.

I have been trying to figure out why default wolfcrypt support for HMAC is not being built in, even after adding NO_HMAC to user_settings.h Thanks for pointing out that there is another NO_HMAC later on. Now it should be all working.

File: include/user_settings.h [hunk 1]
```diff
@@ -122,9 +126,7 @@
 #endif
 
 /* Disables - For minimum wolfCrypt build */
-#define NO_AES
```
Comment (danielinux) [inline include/user_settings.h:125]: Use #ifndef WOLFTPM instead of removing the line

File: include/user_settings.h [hunk 2]
```diff
@@ -122,9 +126,7 @@
...
 #define NO_CMAC
-#define NO_HMAC
```
Comment (danielinux) [inline include/user_settings.h:127]: Use #ifndef WOLFTPM instead of removing the line

File: include/user_settings.h (hunk 1 above)
Comment (tomoveu) [reply include/user_settings.h:125]: good point

File: include/user_settings.h (hunk 2 above)
Comment (tomoveu) [reply include/user_settings.h:127]: thanks, will do

**Review by danielinux — APPROVED** [review]
All non-regression tests are now passing.

**Pull request merged**